addons/web_editor/static/src/js/wysiwyg/widgets/media_dialog.js

```diff
@@ -8,28 +8,39 @@ var Dialog = require('wysiwyg.widgets.Dialog');
 var _t = core._t;
 
 /**
- * MediaDialog widget. Lets users change a media, including uploading a
- * new image, font awsome or video and can change a media into an other
- * media.
+ * Lets the user select a media. The media can be existing or newly uploaded.
+ *
+ * The media can be one of the following types: image, document, video or
+ * font awesome icon (only existing icons).
+ *
+ * The user may change a media into another one depending on the given options.
  */
 var MediaDialog = Dialog.extend({
     template: 'wysiwyg.widgets.media',
     xmlDependencies: Dialog.prototype.xmlDependencies.concat(
         ['/web_editor/static/src/xml/wysiwyg.xml']
     ),
     events: _.extend({}, Dialog.prototype.events, {
-        'click a[data-toggle="tab"]': '_onTabChange',
+        'click #editor-media-image-tab': '_onClickImageTab',
+        'click #editor-media-document-tab': '_onClickDocumentTab',
+        'click #editor-media-icon-tab': '_onClickIconTab',
+        'click #editor-media-video-tab': '_onClickVideoTab',
     }),
     custom_events: _.extend({}, Dialog.prototype.custom_events || {}, {
         save_request: '_onSaveRequest',
     }),
 
     /**
      * @constructor
+     * @param {Element} media
      */
     init: function (parent, options, media) {
-        var self = this;
-        options = options || {};
+        var $media = $(media);
+
+        options = _.extend({}, options);
```

Comment:
@qsm-odoo I added this changed this and the lines below too

Reply:
Note that you could have `clone` rather than this here, but it is fine like that ;)

Reply:
I didn't know about `clone` but I just had a look, apparently it doesn't transform `undefined` to `{}` so `||` would be needed in that case. So in total that's 1 less char. Also `clone` calls `extend` internally: 
https://github.com/jashkenas/underscore/blob/master/underscore.js#L1165

Reply:
Yes, but we don't use underscore for performance but for readability ;) But anyway that was just to let you know this exists, I am fine with `extend`